src/groupByKey.ts

```diff
@@ -1,7 +1,24 @@
-type GroupsMap<T> = {
+export interface GroupsMap<T> {
```

Comment:
I like to see this style in JS, didn't know we could do that. Reminds me of C langs